neuraxle/base.py

```diff
@@ -1764,24 +1764,46 @@ def teardown(self) -> 'BaseStep':
 
         return self
 
-    def patch_missing_names(self, steps_as_tuple: List) -> NamedTupleList:
+    def _wrap_non_base_steps(self, steps_as_tuple: List) -> NamedTupleList:
         """
-        Make sure that each sub step as a unique name, and add a name to the sub steps that don't have one already.
+        If some steps are not of type BaseStep, we'll try to make them of this type. For instance, sklearn objects
+        will be wrapped by a SKLearnWrapper here.
 
-        :param steps_as_tuple: steps as tuple
-        :type steps_as_tuple: NamedTupleList
-        :return:
+        :param steps_as_tuple: a list of steps or of named tuples of steps (e.g.: NamedTupleList)
+        :return: a NamedTupleList
         """
-        names_yet = set()
-        patched = []
-        for step in steps_as_tuple:
+        # TODO: document more the type of the `steps as tuple`.
 
+        wrapped = []
+        for step in steps_as_tuple:
+            class_name = None
             if isinstance(step, tuple):
                 class_name = step[0]
                 step = step[1]
-            else:
+
+            if isinstance(step, BaseEstimator):
+                import neuraxle.steps.sklearn
+                step = neuraxle.steps.sklearn.SKLearnWrapper(step)
```

Comment:
I don't like my own code for these two lines of circular import here. But that works.

Reply:
@arita37 all elements must have fit, fit_transform, and transform. there is a default implementations for all methods as well. there is mixins for default implementations  see : https://www.neuraxio.com/en/neuraxle/stable/examples/non_fittable_mixin.html#sphx-glr-examples-non-fittable-mixin-py

Reply:
@arita37, do you instead mean that the [TensorFlow estimators](https://www.tensorflow.org/api_docs/python/tf/estimator/Estimator) does not have fit or transform methods? 

This will be fixed when we will build wrappers like the scikit-learn wrappers.